src/libultra_code/osReadMempak.c

```diff
@@ -0,0 +1,57 @@
+#include <ultra64.h>
+#include <global.h>
+
+#include <ultra64/controller.h>
+
+s32 osReadMempak(OSMesgQueue* ctrlrqueue, s32 ctrlridx, u16 addr, PIF_mempak_data_t* data) {
+    s32 ret;
+    s32 i;
+    u8* bufptr;
+    s32 read_try_count = 2;
+
+    __osSiGetAccess();
+    do {
+        if ((_osCont_lastPollType != 2) || (ctrlridx != D_80134D20)) {
+            bufptr = &pifMempakBuf.bytes[0];
+            _osCont_lastPollType = (u8)2U;
```

Comment:
I suggest removing all these (u8) casts and `U` values that are in this file, unless they are actually necessary.